### PR TITLE
Add Default case for regmgr_call.

### LINKED ISSUE #332: Syscall 532, Implement Default
Default would allow 0x27 and any types past 0x36 to work.

### PR DESCRIPTION
**Pull request by VocalFan**
Added Default case for regmgr_call, allowing types 0x27 and any type past 0x36 to work. I implemented it in a way that it won't Default any values that DO have a special case (0x0-> 0x26 and 0x28 -> 0x36) are special cases, so I used (0x27 | 0x37..=u32::max)

Fixes #332

### CODE REVIEW

File: src/kernel/src/syscalls/mod.rs [hunk 1]
```diff
@@ -243,6 +243,9 @@ impl Syscalls {
                     todo!("regmgr_call with matched key = {key:#x}");
                 }
             }
+            // Pass default value for any values that aren't given a unique case
+            0x27 | 0x37..=u32::MAX => -2146631143,
+            // Any missing cases TODO
```
Comment (ultimaweapon) [inline src/kernel/src/syscalls/mod.rs:248]:
```c
    case 0x37:
        if (DAT_ffffffffa2830830 == 0) {
LAB_ffffffffa08df0f3:
            key = 0x800d0201;
        }
        else {
            if ((DAT_ffffffffa2830834 == 0) || (DAT_ffffffffa2830800 == 1)) {
                key = FUN_ffffffffa08ced20();
                goto LAB_ffffffffa08df4bf;
            }
LAB_ffffffffa08df0dd:
            key = 0x800d020d;
            if ((int)DAT_ffffffffa2830834 < 0) {
                key = DAT_ffffffffa2830834;
            }
        }
LAB_ffffffffa08df4cc:
        rVar8 = 0;
LAB_ffffffffa08df4d6:
        FUN_ffffffffa08eba50(5,rVar8,key & 0xffff);
        break;
```

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/syscalls/mod.rs (hunk 1 above)
Comment (VocalFan) [reply src/kernel/src/syscalls/mod.rs:248]: Damn, I was one off XD

**Comment by VocalFan** [comment]
Not sure where you got 0x37 though, as that wasn't in sys_regmgr_call

**Comment by VocalFan** [comment]
Seeing as:

```Cpp
        case 0x36uLL:
          v33 = sceRegMgrPrivateStorageIsNvsChange();
          break;
        default:
          v33 = -2146631143;
          break;
      }
```

**Comment by ultimaweapon** [comment]
If you take those from the decompiled it is an old one.

File: src/kernel/src/syscalls/mod.rs [hunk 2]
```diff
@@ -243,6 +243,9 @@ impl Syscalls {
                     todo!("regmgr_call with matched key = {key:#x}");
                 }
             }
+            // Pass default value for any values that aren't given a unique case
+            0x27 | 0x38..=u32::MAX => -2146631143,
+            // Any missing cases TODO
```
Comment (ultimaweapon) [inline src/kernel/src/syscalls/mod.rs:248]:
```c
    case 0x38:
        if (DAT_ffffffffa2830830 == 0) goto LAB_ffffffffa08df0f3;
        if ((DAT_ffffffffa2830834 != 0) && (DAT_ffffffffa2830800 != 1)) goto LAB_ffffffffa08df0dd;
        key = FUN_ffffffffa08d04d0();
        goto LAB_ffffffffa08df4bf;
```

**Review by ultimaweapon — CHANGES_REQUESTED** [review]

File: src/kernel/src/syscalls/mod.rs (hunk 2 above)
Comment (VocalFan) [reply src/kernel/src/syscalls/mod.rs:248]: Stop going one by one because the FW leak only goes to 36 XD

Comment (ultimaweapon) [reply src/kernel/src/syscalls/mod.rs:248]: If you mean the one that you send it to me it is not a source leak but an old binary leak that was decompiled by IDA Pro. You should follow the most recent one that you have, which is 9.0; and use the leaked one as a supplement.

Comment (VocalFan) [reply src/kernel/src/syscalls/mod.rs:248]: Ah, I see 0x39 is the highest so far.

**Comment by VocalFan** [comment]
Fixed it by looking at and searching in 9.00's regmgr_call. Highest value was 0x39

**Comment by VocalFan** [comment]
Well, that confirms the hex is indeed -2146631143

**Review by ultimaweapon — APPROVED** [review]

**Pull request merged**